src/OpenSpaceToolkit/Astrodynamics/Flight/Profile.cpp

```diff
@@ -425,6 +480,28 @@ Vector3d Profile::ComputeTargetDirectionVector(const State& aState, const Trajec
     return (targetPositionCoordinates - satellitePositionCoordinates).normalized();
 }
 
+Vector3d Profile::ComputeTargetVelocityVector(const State& aState, const ostk::astrodynamics::Trajectory& aTrajectory)
+{
+    if (!aTrajectory.isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Trajectory");
+    }
+
+    const Transform ITRF_GCRF_transform = Frame::ITRF()->getTransformTo(Frame::GCRF(), aState.accessInstant());
+
+    const State slidingTargetState_GCRF = aTrajectory.getStateAt(aState.accessInstant());
+    const State slidingTargetState_ITRF = slidingTargetState_GCRF.inFrame(Frame::ITRF());
+
+    const Vector3d relativePositionDirection =
+        (slidingTargetState_GCRF.getPosition().getCoordinates() - aState.getPosition().getCoordinates()).normalized();
+    const Vector3d relativeVelocityDirection =
+        ITRF_GCRF_transform.applyToVector(slidingTargetState_ITRF.getVelocity().getCoordinates()).normalized();
+
+    const Vector3d relativeNormalDirection = relativePositionDirection.cross(relativeVelocityDirection).normalized();
+
+    return relativeNormalDirection.cross(relativePositionDirection);
+}
```

Comment:
now that we properly construct the trajectory velocity in gcrf at trajectory construction time, why can't we just do this computation the same way as the ComputeTargetDirectionVector method? Where you do
`targetVelocityCoordinates - satelliteVelocityCoordinates).normalized()`?

I believe you'd get the same result. Here's youre doing the cross product of the position and velocity and then of the position and normal which gives you back a vector aligned with the original velocity direction

Reply:
The position and velocity are not orthogonal necessarily. r x v -> Normal, which is orthogonal to the plane of r & v, and then crossing r x n gets us an orthogonal vector to the r & n plane which is not necessarily aligned with the velocity direction.